custom_methods.rb

```diff
@@ -0,0 +1,145 @@
+# rubocop:disable Style/CaseEquality
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def change_self
+    return Array self if self.class == Range
+
+    self
+  end
+
+  def my_each
+    my_array = change_self
+    return to_enum unless block_given?
+
+    i = 0
+    while i < my_array.length
+      yield my_array[i]
+      i += 1
+    end
+    my_array
+  end
+
+  def my_each_with_index
+    my_array = change_self
+    return to_enum unless block_given?
+
+    i = 0
+    while i < my_array.length
+      yield my_array[i], i
+      i += 1
+    end
+    my_array
+  end
+
+  def my_select
+    my_array = change_self
+    return to_enum unless block_given?
+
+    arr = []
+    my_array.my_each { |x| arr.push(x) if yield x }
+    arr
+  end
+
+  def my_all?(word = nil)
+    my_array = change_self
+    if block_given?
+      my_array.my_each { |item| return false unless yield item }
+      true
+    elsif word
+      my_array.my_each { |item| return false unless word === item }
+      true
+    else
+      my_all? { |item| item }
+    end
+  end
+
+  def my_any?(regex = false)
+    my_array = change_self
+    if block_given?
+      my_array.my_each { |item| return true if yield item }
+      false
+    elsif regex
+      my_array.my_each { |item| return true if regex === item }
+      false
+    else
+      my_any? { |item| item }
+    end
+  end
+
+  def my_none?(regex = false)
+    my_array = change_self
+    if block_given?
+      my_array.my_each { |item| return false if yield item }
+      true
+    elsif regex
+      my_array.my_each { |item| return false if regex === item }
+      true
+    else
+      my_none? { |item| item }
+    end
+  end
```

Comment:
#### my_none

[OPTIONAL]

I think if we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

```
  def my_none?(arg = nil, &block)
    !my_any?(arg = nil,&block)
  end
```

To make this work we have to use block.call(element) instead of yield(element) in my_any?